api/controllers/courseController.js

```diff
@@ -0,0 +1,119 @@
+import Course from "../models/course-model.js";
+import User from "../models/user-model.js";
+import Lecture from "../models/lecture-model.js";
+const createCourse = async (req, res) => {
+  const instructorId = req.user.sub;
+  const { title, description, price, imageUrl } = req.body;
+
+  try {
+    // Ensure the instructor exists and is valid
+    const instructor = await User.findOne({
+      where: { id: instructorId, role: "instructor" },
+    });
+    if (!instructor) {
+      return res
+        .status(403)
+        .json({ message: "Instructor not found or invalid." });
+    }
+
+    // Create the course
+    const course = await Course.create({
+      title,
+      description,
+      instructorId,
+      imageUrl,
+      price,
+    });
+
+    res.status(201).json(course);
+  } catch (error) {
+    console.error(error);
+    res.status(500).json({ error: "Error creating course." });
+  }
+};
+
+const getAllCourses = async (req, res) => {
+  try {
+    // Fetch all courses along with their instructor details
+    const courses = await Course.findAll({
+      include: {
+        model: User,
+        as: "instructor",
+        attributes: ["id", "name", "email"], // Select specific fields for the instructor
+      },
+      attributes: ["id", "title", "description", "imageUrl", "createdAt"], // Select specific fields for the course
+    });
+
+    if (!courses.length) {
```

Comment:
I don't mind throwing a 404 here, but what is the frontend expecting? 

It's probably expecting an empty array if there are no courses. Please ensure that the frontend is not showing an error if there are no courses, it should display "No courses found" or something like this, Not an error!